src/tests/conformance/handshake.rs

```diff
@@ -18,9 +18,10 @@ async fn handshake_when_node_receives_connection() {
         listener_ip: Some("127.0.0.1".parse().unwrap()),
         ..Default::default()
     };
-
-    // TODO: replace with a connection from the node to signal readiness.
-    tokio::time::sleep(CONNECTION_TIMEOUT).await;
+    wait_until!(
```

Comment:
We might want to drop the connection on success, just so we don't have a persistant connection during tests?

Reply:
Isn't it dropped when inside `loop` ends?

Reply:
Yep, my bad, the stream will get dropped 👍 . That being said, I think this call should probably not be in the test but in the node setup code (as with the Zcash impl).

Reply:
Note that in the zcash impl, we do the opposite, we wait for a connection from the node—I currently don't see a problem with your approach, just that it may cause breakage on the node's side if it can't handle too many conns being opened (1 per ms if I recall currectly in the `wait_until!` impl).

Reply:
> Yep, my bad, the stream will get dropped
Well, it _should_ but `TcpStream` doesn't actually implement `Drop`. The read half is usually dropped when the write half is but since we don't actually split the stream, it might be best to call `shutdown` to be safe.

Reply:
Done